client/src/app/hooks/table-controls/getHubRequestParams.ts

```diff
@@ -57,3 +57,35 @@ export const serializeRequestParamsForHub = (
   serializePaginationRequestParamsForHub(deserializedParams, serializedParams);
   return serializedParams;
 };
+
+interface HubRequestParamsQuery {
+  /**
+   * The maximum number of entries to return.
+   *
+   * Zero means: no limit
+   */
+  limit?: number;
+  /**
+   * The first item to return, skipping all that come before it.
+   *
+   * NOTE: The order of items is defined by the API being called.
+   */
+  offset?: number;
+  q?: string;
+  sort?: string;
+}
+
+/**
+ * Like serializeRequestParamsForHub but returns a plain object instead of URLSearchParams.
+ */
+export const requestParamsQuery = (
+  deserializedParams: HubRequestParams
+): HubRequestParamsQuery => {
+  const params = serializeRequestParamsForHub(deserializedParams);
+  return {
+    limit: Number(params.get("limit")),
+    offset: Number(params.get("offset")),
```

Comment:
```
Number(undefined) = NaN
```

We should use the same logic used for `q` and `sort` where the code validates a possible undefined value of `params.get("")`

I know the code the currently calls this method always send `limit` & `offset` but better to write safe code.

Reply:
I'll do a full implementation that avoids reusing the serializeRequestParamsForHub hack, that way it can be more efficient and more typesafe.